identity-account/src/updates/update.rs

```diff
@@ -61,56 +58,41 @@ pub(crate) async fn create_identity(
     UpdateError::InvalidMethodType(method_type)
   );
 
-  let generation = Generation::new();
-  let fragment: String = format!("{}{}", DEFAULT_UPDATE_METHOD_PREFIX, generation.to_u32());
+  let account_id: AccountId = AccountId::random();
 
-  let location: KeyLocation = KeyLocation::new(method_type, fragment, generation);
+  let tmp_location: KeyLocation = KeyLocation::random(setup.key_type);
 
-  let keypair: KeyPair = match &setup.method_secret {
-    None => KeyPair::new(KeyType::Ed25519)?,
-    Some(MethodSecret::Ed25519(private_key)) => {
-      ensure!(
-        private_key.as_ref().len() == ed25519::SECRET_KEY_LENGTH,
-        UpdateError::InvalidMethodSecret(format!(
-          "an Ed25519 private key requires {} bytes, found {}",
-          ed25519::SECRET_KEY_LENGTH,
-          private_key.as_ref().len()
-        ))
-      );
-      KeyPair::try_from_private_key_bytes(KeyType::Ed25519, private_key.as_ref())?
-    }
-    _ => {
-      return Err(UpdateError::InvalidMethodSecret("expected None or Ed25519 private key".to_owned()).into());
-    }
-  };
+  if let Some(inner_method_secret @ MethodSecret::Ed25519(_)) = setup.method_secret {
+    insert_method_secret(store, account_id, &tmp_location, setup.key_type, inner_method_secret).await?;
+  } else {
+    store.key_generate(&account_id, &tmp_location).await?;
+  }
```

Comment:
This case is technically handled a few lines above with:

```rust
// The method type must be able to sign document updates.
  ensure!(
    IotaDocument::is_signing_method_type(method_type),
    UpdateError::InvalidMethodType(method_type)
  );
```

Would you prefer handling the case of other `MethodSecret`s explicitly anyway?

Reply:
That case is not handled. You can reproduce the problem by passing `MethodType::Ed25519` with `MethodSecret::X25519`, it just generates a new key and ignore the `X25519` private key (I ran into that case myself).

Probably made irrelevant with #764 since I removed `MethodSecret` here altogether.